Do not merge - blocked: TS Models (#2281)
mongodb/docs-realm: source/sdk/react-native/install.txt

            .. include:: /includes/react-native-run-the-app.rst

.. _react-native-typescript-support:

TypeScript Support (Optional)
-----------------------------

If you are using `React Native version 0.60 or greater
<https://reactnative.dev/versions>`_, you can declare a Realm Object Model using

> TypeScript property types declarations on classes. The Realm JS SDK will derive

Reviewer: I asked this in the Node.js SDK content, and I think it's valuable here too. Do we refer to the "Realm JS SDK" elsewhere? In this instance, shouldn't we say "React Native SDK"? I know Node and React Native use the same source, but we present them as two different SDKs.
```suggestion
TypeScript property types declarations on classes. The Realm React Native SDK will derive
```

> the schema at runtime from the types declared in your TS source files. This

Reviewer: We haven't abbreviated TypeScript to TS before this. I recommend spelling it out or using `.ts` here.
```suggestion
the schema at runtime from the types declared in your `.ts` source files. This
```

allows you to use TypeScript syntax without needing to explicitly define a
schema.

> To do so, follow the steps below to install and configure:
> 
> -  the :npm:`Realm Babel Plugin <package/@realm/babel-plugin>`
> -  the :npm:`@babel/plugin-proposal-decorators <package/@babel/plugin-proposal-decorators>`

Reviewer: We generally don't use sentence fragments in lists to complete an introductory sentence. It's harder to read than being more explicit.
```suggestion
To implicitly declare a Realm Object Model, follow the steps below to install and configure the :npm:`Realm Babel Plugin <package/@realm/babel-plugin>` and the :npm:`@babel/plugin-proposal-decorators <package/@babel/plugin-proposal-decorators>`.
```
"Realm Object Model" should be "Realm Object Schema" if we're going with the model -> schema suggestion.

Although we recommend installing and configuring
``@babel/plugin-proposal-decorators`` as detailed below, it is only required if
you intend to use :ref:`decorators  <react-native-ts-decorators>` in your
classes. 

.. procedure::

   .. step:: Enable TypeScript

      .. include:: /includes/react-native-enable-typescript.rst

   .. step:: Install the Realm Babel Plugin and Babel Plugin Proposal Decorators Package
      
      Run the following command to add them to your `dev dependencies
      <https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file>`_:

      .. code-block:: shell
         
         npm install --save-dev @realm/babel-plugin @babel/plugin-proposal-decorators

   .. step:: Add the Plugins to Your Babel Config

>       Add the following to your ``babel.config.js`` file to apply the `plugins <https://babeljs.io/docs/en/plugins>`_

Reviewer: ```suggestion
      Add the following to your ``babel.config.js`` file to apply the `plugins <https://babeljs.io/docs/en/plugins>`_:
```

      .. code-block:: javascript

         const presets = [ ... ];
         module.exports = { 
            presets,
            plugins: [
               '@realm/babel-plugin',
               ['@babel/plugin-proposal-decorators', { legacy: true }],  // Optional, unless you're using decorators
            ],
         };
   .. step:: Enable Decorators in your TS Config File (Optional)

>       Enable experimental decorators in your `tsconfig.json <https://www.typescriptlang.org/docs/handbook/tsconfig-json.html>`_ within your ``compilerOptions``:

Reviewer: ```suggestion
      Enable experimental decorators in your `tsconfig.json <https://www.typescriptlang.org/docs/handbook/tsconfig-json.html>`_ within your ``compilerOptions`` file:
```

      .. code-block:: json

         {
            "compilerOptions": {
               "experimentalDecorators": true
            },
         }

   .. step:: Restart the Package Cache

>       If you're using TypeScript, you must restart the package cache to work with the newly applied plugin in your app. For most React Native applications, you can run the command:

Reviewer: ```suggestion
      If you're using TypeScript, you must restart the package cache to work with the newly-applied plugin in your app. For most React Native applications, you can run the command:
```
